Documentation/design-docs/DotNetCore-SharedPackageStore.md

```diff
@@ -0,0 +1,219 @@
+
+If `--optimize` is specified, we would precompile all the managed assets to native code in a temp folder before copying to the output folder. If crossgen is used, it would be the one acquired in the closure of the `Microsoft.NETCore.App` specified by the `--framework-version` option. Also, if no `--output` folder is specified, then the default is `~/.dotnet` or `%USERPROFILE%\.dotnet\`. The output asset files will be present in the following layout: `$HOME/.dotnet/packages/{tfm}/{package-name}/{package-version}/{asset-path}`.
+
+The output folder will be consumed by the runtime by adding to the `DOTNET_SHARED_PACKAGES_ROOTS` environment variable. See probe precedence below.
+
+# Building apps with shared packages
+
+The current mechanism to build applications that share assemblies is by not specifying a RID in the project file. Then, a portable app model is assumed and assemblies that are part of Microsoft.NETCore.App are found under the `dotnet` install root. With shared package store, applications have the ability to filter any set of packages from their publish output. Thus the decision of a portable or a standalone application is not made at the time of project authoring but is instead done at publish time. 
+
+## Project Authoring
+We will by default treat `Microsoft.NETCore.App` as though `type: platform` is always specified, thus requiring no explicit RID specification by the user. It will be an `ERROR` to specify a RID in the csproj file using the `<RuntimeIdentifier/>` tag.
+
+## dotnet restore
+
+Because RIDs are not available until publish time, the application is treated as though it is a present day portable app and a regular restore is performed.
+
+## dotnet build
+
+`dotnet build` should treat any project as though it is a portable app and produce `runtimeconfig.json` with the `Microsoft.NETCore.App` as the framework. In addition, the `runtimeconfig.json` file should also specify the TFM field as:
+
+```json
+"runtimeOptions": {
+    "tfm": "netcoreapp2.0",
+    "framework": {
+        "version": "2.0.0",
+        "name": "Microsoft.NETCore.App"
+    }
+}
+```
+Note that this is different from current behavior of `dotnet run` for an application that specifies the `<RuntimeIdentifier/>` tag in the csproj.
+**Current Behavior:** Picks `M.N.A` assemblies out of the NuGet cache without taking advantage of optimizations available from the shared `Microsoft.NETCore.App`.
+**New Behavior:** Picks `M.N.A` assemblies from the shared framework and the rest of them from the shared package store or the NuGet cache.
+
+`dotnet build` can take advantage of the `refs` folder available at the `packages/install/` folder from the `dotnet` root directory enabling the offline-restore-build scenario. Although we are designing to augment `dotnet build` in the future regarding the reference assemblies, for the scope of this work we'll focus only on runtime assemblies.
```

Comment:
Should we be discussing `refs` at all in this doc since this work is out of scope?